openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -839,3 +841,23 @@ def get_neutron_endpoint(self):
 
     def get_networks(self):
         return self._api.get_networks()
+
+    def _get_keystone_server_url(self, instance_config):
+        keystone_server_url = instance_config.get("keystone_server_url")
+        if keystone_server_url:
+            return keystone_server_url
+
+        openstack_config_file_path = instance_config.get("openstack_config_file_path")
+        openstack_cloud_name = instance_config.get("openstack_cloud_name")
+        with open(openstack_config_file_path, 'r') as stream:
```

Comment:
You should be able to use the `OpenStackConfig` class instead of reimplementing the yaml parsing.